modules/loki-entry-out-of-order-errors.adoc

```diff
@@ -0,0 +1,75 @@
+// Module is included in the following assemblies:
+// cluster-logging-loki.adoc
+
+:_content-type: PROCEDURE
+[id="loki-entry-out-of-order-messages_{context}"]
+= Troubleshooting Loki entry out of order errors
+
+If your Fluentd instance forwards a large block of messages that exceeds the rate limit to a Loki logging system, Loki generates `entry out of order` errors. You can fix this issue by modifying the `LokiStack` custom resource (CR).
+
+[IMPORTANT]
+====
+The `LokiStack` CR is not available on Grafana-hosted Loki. This topic does not apply to Grafana-hosted Loki servers.
+====
+
+.Conditions
+
+* The `ClusterLogForwarder` custom resource (CR) is configured to forward logs to Loki.
+
+* Your system sends a block of messages that is larger than 2 MB to Loki. For example:
++
+[source,text]
+----
+"values":[["1630410392689800468","{\"kind\":\"Event\",\"apiVersion\":\
+.......
+......
+......
+......
+\"received_at\":\"2021-08-31T11:46:32.800278+00:00\",\"version\":\"1.7.4 1.6.0\"}},\"@timestamp\":\"2021-08-31T11:46:32.799692+00:00\",\"viaq_index_name\":\"audit-write\",\"viaq_msg_id\":\"MzFjYjJkZjItNjY0MC00YWU4LWIwMTEtNGNmM2E5ZmViMGU4\",\"log_type\":\"audit\"}"]]}]}
+----
+
+* When you enter `oc logs -c collector`, the Fluentd logs in your OpenShift Logging cluster show the following error messages:
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded (limit: 8388608 bytes/sec) while attempting to ingest '2140' lines totaling '3285284' bytes
+----
++
+[source,text]
+----
+429 Too Many Requests Ingestion rate limit exceeded' or '500 Internal Server Error rpc error: code = ResourceExhausted desc = grpc: received message larger than max (5277702 vs. 4194304)'
+----
+
+* After you open the logs on the Loki server, `entry out of order` messages similar to the following are displayed:
++
+[source,text]
+----
+,\nentry with timestamp 2021-08-18 05:58:55.061936 +0000 UTC ignored, reason: 'entry out of order' for stream:
+
+{fluentd_thread=\"flush_thread_0\", log_type=\"audit\"},\nentry with timestamp 2021-08-18 06:01:18.290229 +0000 UTC ignored, reason: 'entry out of order' for stream: {fluentd_thread="flush_thread_0", log_type="audit"}
```

Comment:
If we have specified vector as collector in CL instance, warning messages should be shown as per vector and not fluentd.

Reply:
Could you provide new error message examples, please?

Reply:
`2023-08-25T16:08:49.301780Z  WARN sink{component_kind="sink" component_id=default_loki_infra component_type=loki component_name=default_loki_infra}: vector::sinks::util::retries: Retrying after error. error=Server responded with an error: 429 Too Many Requests internal_log_rate_limit=true`

Reply:
I will update the context maybe of this section then to talk about "rate limit errors" instead of "entry out of order errors" if that makes more sense?

Reply:
Could it be related to this maybe: https://vector.dev/docs/reference/configuration/sinks/loki/#out_of_order_action ?
It looks like newer versions of Loki support out of order events, so maybe it doesn't throw this error anymore?